tests/phpunit/tests/rest-api/rest-global-styles-controller.php

```diff
@@ -145,17 +160,73 @@ public function test_get_theme_item_invalid() {
 	}
 
 	/**
+	 * @dataProvider data_get_theme_item_invalid_theme_dirname
 	 * @covers WP_REST_Global_Styles_Controller::get_theme_item
+	 * @ticket 54516
 	 */
-	public function test_get_theme_item() {
+	public function test_get_theme_item_invalid_theme_dirname( $theme_dirname ) {
 		wp_set_current_user( self::$admin_id );
-		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/tt1-blocks' );
+		switch_theme( $theme_dirname );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/' . $theme_dirname );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertErrorResponse( 'rest_no_route', $response, 404 );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_get_theme_item_invalid_theme_dirname() {
+		return array(
+			'with |'                 => array( 'my|theme' ),
+			'with +'                 => array( 'my+theme' ),
+			'with {}'                => array( 'my{theme}' ),
+			'with #'                 => array( 'my#theme' ),
+			'with !'                 => array( 'my!theme' ),
+			'multiple invalid chars' => array( 'mytheme-[_(+@)]#! v4.0' ),
+		);
+	}
+
+	/**
+	 * @dataProvider data_get_theme_item
+	 * @covers WP_REST_Global_Styles_Controller::get_theme_item
+	 * @ticket 54516
+	 */
+	public function test_get_theme_item( $theme ) {
+		wp_set_current_user( self::$admin_id );
+		switch_theme( $theme );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/' . $theme );
 		$response = rest_get_server()->dispatch( $request );
 		$data     = $response->get_data();
 		unset( $data['_links'] );
 
-		$this->assertArrayHasKey( 'settings', $data );
-		$this->assertArrayHasKey( 'styles', $data );
+		$this->assertArrayHasKey( 'settings', $data, 'Data does not have "settings" key' );
+		$this->assertArrayHasKey( 'styles', $data, 'Data does not have "styles" key' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_get_theme_item() {
+		return array(
+			'alphabetic chars'   => array( 'mytheme' ),
```

Comment:
https://github.com/WordPress/wordpress-develop/pull/2034/files#r768865447

Reply:
Testing would need to be done on cross-platforms (i.e. Windows, Mac, Linux, etc) to determine if non-latin characters and/or emojis are valid in filesystem directory names.